src/hooks/useWalletsHiddenBalances.ts

```diff
@@ -18,17 +18,18 @@ const getHiddenAssetBalance = ({
   address,
   nativeCurrency,
   connectedToHardhat,
+  data,
 }: {
   address: Address;
   nativeCurrency: NativeCurrencyKey;
   connectedToHardhat: boolean;
+  data?: UserAssetsResult;
 }) => {
   const { hiddenAssets } = userAssetsStore.getState(address);
-  const assetData = queryClient.getQueryData<UserAssetsResult>(
-    userAssetsQueryKey({ address, currency: nativeCurrency, connectedToHardhat })
-  );
+  const assetData =
```

Comment:
couldn't you always reliably get the query data from the cache, instead of sometimes passing it in as a param?